### ISSUE TITLE
Confusing info about joining a group

### ISSUE THREAD

**Issue by mohanboddu** [fedora-infra/noggin#192]
When I go to `admins` group, it says "To join this group, contact a group sponsor.", but there are no sponsors listed.

So, does every group should have at least one sponsor?

1. If not, the above statement should change to
"To join this group, contact a group sponsor (`or members or admin`)" assuming one of them has the ability to sponsor.

2. If the group has at least one sponsor, then something went wrong in creation of that group.

**Comment by ryanlerch** [comment]
@abompard Only IPA admins are able to add groups, and add / remove sponsors, right? and this is done manually, not through noggin, right?

So do we add something in theIPA config to say a group has to have at least one sponsor? or its it SOP for when a admin adds a group?

**Comment by abompard** [comment]
Yes, at the moment only IPA admins can do that. I think it should be a SOP, but maybe noggin can handle the situation better and say something like "no sponsors, this should not happen, please come back later or open a ticket" or something like that. No ?

**Comment by nirik** [comment]
Just FYI, currently in fas: 

group admins can elevate users to sponsor or downgrade them to users or remove them from the group. 

sponsors can only sponsor new users or remove users. 

Groups can be 'invite only' which means a sponsor or admin has to add them, or they can not be invite only in which case the user can request joining the group, but a sponsor or admin still has to approve them, they are just in a pending state. 

Groups always have an admin (the first person in the group when it's made).

There is a special group in fas called 'accounts' that has super permissions and can add/remove anyone from any group or promote/demote people from user/sponsor/admin. 

So, thats fas, we don't have to match it 100%, but thats what we have now. :) 

Perhaps we could add another group attribute called 'join information' and it points to a url with information on how to join the group?
Do the 'user' 'sponsor' 'admin' labels/abilities make any sense in the noggin world?

**Comment by abompard** [comment]
> group admins can elevate users to sponsor or downgrade them to users or remove them from the group.

According to the people who designed the system, this was explicitely marked as a non-goal for the web UI, and should be done via FreeIPA's UI or command line, see #47

> sponsors can only sponsor new users or remove users.

I think that's how it works at the moment.

> Groups can be 'invite only' which means a sponsor or admin has to add them, or they can not be invite only in which case the user can request joining the group, but a sponsor or admin still has to approve them, they are just in a pending state.

According to #50, users can't request to join a group from the web UI, they have to contact a sponsor. It effectively boils down to just having the invite-only process.

> Groups always have an admin (the first person in the group when it's made).

Here members and sponsors are two different lists, nothing in the web UI forces a group to have a sponsor, and sponsors are not necessarily members of the group. Since groups are created in FreeIPA and not in noggin, I'm not sure we can force groups to have a sponsor. At least I don't remember it being a requirement.

> There is a special group in fas called 'accounts' that has super permissions and can add/remove anyone from any group or promote/demote people from user/sponsor/admin.

Those people would be FreeIPA admins.

> So, thats fas, we don't have to match it 100%, but thats what we have now. :)

Good to know, thanks. However, the requirements we got described different processes, I hope that's alright. Some things can still be added if necessary (depending on feasability in FreeIPA and security policy), but I'd like to check with Ricky, Patrick and Christian before we understake something that's too different from the original design.

> Perhaps we could add another group attribute called 'join information' and it points to a url with information on how to join the group?

There has been suggestions on extended group attributes (see fedora-infra/freeipa-fas#7) but it hasn't been planned nor implemented. We also would need to think of how this "how to join" text should be updated, be it by sponsors in noggin or by admins in FreeIPA.

> Do the 'user' 'sponsor' 'admin' labels/abilities make any sense in the noggin world?

Yes, those labels map to membermanagers and admins in FreeIPA I believe.

**Comment by nirik** [comment]
ok, so to clarify the way it works currently: 
* membermanagers can add new users to the groups they are membermanagers of right?
* but only via a ipa cli? (Thats not going to be great... we don't want any membermanager to have to have direct access to the ipa servers, which means they will need shell on some internal machine)
* only admins can add new membermanagers via cli?
* membermanagers are per group?
* when we migrate from fas we can make all existing sponsors membermanagers of the groups they are currently sponsors in?

**Comment by relrod** [comment]
@nirik The way it was planned to work last I knew/before I transferred was: Membermanagers can add new users to the group (and remove them from the group) via the UI (they don't need CLI access). But only admins (think: people in fas's `accounts` group) can control the membermanagers of groups.

So adding a sponsor to a group will require an infra ticket, but adding a user to a group will just be a UI button click for sponsors.

And yes, when I turned over fas2ipa, it did indeed retain sponsorship information. I'm assuming that hasn't changed.
